packages/fluentui/react-builder/src/config.tsx

```diff
@@ -0,0 +1,717 @@
+  }
+
+  return omitChildWithUuid(tree, uuid);
+};
+
+export const jsonTreeCloneElement = (tree: JSONTreeElement, element: any): JSONTreeElement => {
+  const result = _.transform(element, (acc, value, key) => {
+    if (_.isPlainObject(value)) {
+      acc[key] = jsonTreeCloneElement(tree, value);
+    } else if (Array.isArray(value)) {
+      acc[key] = jsonTreeCloneElement(tree, value);
+    } else {
+      acc[key] = value;
+    }
+  }) as any;
+
+  if (_.isPlainObject(result) && result.$$typeof === 'Symbol(react.element)' && result.uuid) {
+    result.uuid = getUUID();
+  }
+  return result;
+};
+
+/**
+ * Displays a knob with the ability to switch between data `types`.
+ */
+export const MultiTypeKnob: React.FunctionComponent<{
+  label: string;
+  types: ('boolean' | 'number' | 'string' | 'literal')[];
+  value: any;
+  onChange: (value: any) => void;
+  literalOptions: string[];
+}> = ({ label, types, value, onChange, literalOptions }) => {
+  const valueType = typeof value;
+  const defaultType = valueType !== 'undefined' ? valueType : types[0];
+  const [type, setType] = React.useState(defaultType);
+  const knob = knobs[type];
+  const handleChangeType = React.useCallback(
+    e => setType(e.target.value), // @ts-ignore
+    [],
+  );
+
+  // console.log('MultiTypeKnob', { label, value, type, types });
+
+  return (
+    <div style={{ paddingBottom: '4px', marginBottom: '4px', opacity: knob ? 1 : 0.4 }}>
+      <div>
+        {type !== 'boolean' && <label>{label} </label>}
+        {types.length === 1 ? (
+          <code style={{ float: 'right' }}>{type}</code>
+        ) : (
+          types.map(t => (
+            <button key={t} onClick={() => handleChangeType(t)}>
+              {t}
+            </button>
+          ))
+        )}
+      </div>
+      {knob && knob({ options: literalOptions, value, onChange })}
+      {type === 'boolean' && <label> {label}</label>}
+    </div>
+  );
+};
+
+export const knobs = {
+  boolean: ({ value, onChange }) => (
+    <input type="checkbox" checked={!!value} onChange={e => onChange(!!e.target.checked)} />
+  ),
+
+  number: ({ value, onChange }) => (
+    <input
+      style={{ width: '100%' }}
+      type="number"
+      value={Number(value)}
+      onChange={e => onChange(Number(e.target.value))}
+    />
+  ),
+
+  string: ({ value, onChange }) => (
+    <input style={{ width: '100%' }} value={String(value)} onChange={e => onChange(e.target.value)} />
+  ),
+
+  literal: ({ options, value, onChange }) => (
+    <select onChange={e => onChange(e.target.value)} value={value}>
+      {options?.map((
+        opt, // FIXME the optional is workaround for showing `Dialog` props when selected from component tree
+      ) => (
+        <option key={opt} value={opt}>
+          {opt}
+        </option>
+      ))}
+    </select>
+  ),
+
+  ReactText: (value, onChange) => knobs.string({ value, onChange }),
+  'React.ElementType': (value, onChange) => knobs.string({ value, onChange }),
+};
```

Comment:
did you consider handling shorthand as a plain input? It might not work for all cases, but at least it would allow us to add for example `content` to `List.Item`